src/test/java/org/jenkinsci/plugins/envinject/GlobalPropertiesTest.java

```diff
@@ -91,4 +98,62 @@ public void testGlobalPropertiesSetWORKSPACE() throws Exception {
 
         assertEquals(result_testGlobalVariableName, result_testJobVariableName);
     }
+
+    @Test
+    public void testChangeOfGlobalPropertyGetsRecognizedWhenWithoutJobPropertyAndRunOnSlaves() throws Exception {
```

Comment:
@lanwen Indeed. The PR is [WiP]. I thought to review this kind of things. Anyway, thanks so much.